### PR TITLE
Refactor logging in generic classes

### PR DESCRIPTION
**Pull request by andrzejpindor**
**PR Title**
Refactor logging in generic classes

**PR Description**
This will prevent allocating a new copy of logger for each generic type.

Addresses #31340

### CODE REVIEW

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
I assume you searched the rest of the repo for similar patterns?

**Comment by andrzejpindor** [comment]
> I assume you searched the rest of the repo for similar patterns?

Yes, I've searched all classes and partial classes and checked if defined loggers are on classes with generic type.  This PR covers everything I found.

**Comment by BrennanConroy** [comment]
Awesome! Thanks for the PR, I'll merge this when the checks pass.

File: src/Servers/HttpSys/src/RequestProcessing/RequestContextOfT.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Hosting.Server;
 using Microsoft.AspNetCore.Http;
+using Log = Microsoft.AspNetCore.Server.HttpSys.RequestContextLog;
```
Comment (davidfowl) [inline src/Servers/HttpSys/src/RequestProcessing/RequestContextOfT.cs:5]: Love it!

**Comment by davidfowl** [comment]
@andrzejpindor you're on a roll. I hope we see more contributions!

**Pull request merged**